lte/gateway/python/magma/pipelined/app/uplink_bridge.py

```diff
@@ -94,12 +105,44 @@ def initialize_on_connect(self, datapath):
                                                 self.config.uplink_eth_port_name)
         self._install_flow(1000, match, actions)
 
-        # 2.b. All ingress IP traffic for UE mac
-        match = "in_port=%s,ip, dl_dst=%s" % (self.config.uplink_eth_port_name,
-                                              self.config.virtual_mac)
-        actions = "output:%s" % self.config.uplink_patch
+        if self.config.ovs_vlan_workaround:
+            # 2.b. All ingress IP traffic for UE mac
+            match = "in_port=%s,ip, dl_dst=%s, vlan_tci=0x0000/0x1000" % \
+                    (self.config.uplink_eth_port_name,
+                     self.config.virtual_mac)
+            actions = "output:%s" % self.config.uplink_patch
+            self._install_flow(1000, match, actions)
+
+            match = "in_port=%s,ip, dl_dst=%s, vlan_tci=0x1000/0x1000" % \
+                    (self.config.uplink_eth_port_name,
+                     self.config.virtual_mac)
+            actions = "strip_vlan,output:%s" % self.config.dev_vlan_in
+            self._install_flow(1000, match, actions)
+
+            # 2.c. redirect all vlan-out traffic to patch port
+            match = "in_port=%s, dl_dst=%s, ip" % \
+                    (self.config.dev_vlan_out,
+                     self.config.virtual_mac)
+            actions = "output:%s" % self.config.uplink_patch
+            self._install_flow(1000, match, actions)
+        else:
+            # 2.b. All ingress IP traffic for UE mac
+            match = "in_port=%s,ip, dl_dst=%s" % \
+                    (self.config.uplink_eth_port_name,
+                     self.config.virtual_mac)
+            actions = "output:%s" % self.config.uplink_patch
+            self._install_flow(1000, match, actions)
```

Comment:
Seems like a lot of magic numbers, pull out 1000/500/100 into global const priority definitions?